### PR TITLE
Copy docs from scipp

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Docs moved from scipp with minor updates and restructure. There is a lot to improve here to make this more user friendly and readable, but I suggest to leave that for a future step.

### CODE REVIEW

File: docs/about/about.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
+.. _about:
+
+About scippneutron
+==================
+
```
Comment (OwenArnold) [inline docs/about/about.rst:5]: Should say something about scippneutron
```suggestion
``scippneutron`` builds on top of ``scipp`` and provides features specific to handling and reducing data generated in neutron scattering facilities.	
A key example is "unit conversion", e.g., from time-of-flight to energy transfer in an inelastic neutron scattering experiment at a spallation-based neutron source.
```

File: docs/about/contributing.rst [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+Contributing to scippneutron
+============================
+
+Overview
+--------
+
+Contributions, bug reports, and ideas are always welcome.
+The following section outlines the scope of scippneutron.
+If in doubt whether a feature falls within the scope of scippneutron please `ask on github <https://github.com/scipp/scippneutron/issues>`_ before implementing functionality, to reduce the risk of rejected pull requests.
```
Comment (OwenArnold) [inline docs/about/contributing.rst:11]: Question here: We have kept all issues central in scipp, even if relating to `ess` `scippneutron`, `scippwidgets`, etc. This is proposing distributing our issue handling.

File: docs/about/contributing.rst [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+``scippneutron`` shall contain only generic neutron-specific functionality.
+Facility-specific or instrument-specific functionality must not be added.
+Examples of generic functionality that is permitted are 
+
+* Unit conversions, which could be generic for all time-of-flight neutron sources.
+* Published research such as absorption corrections.
+
+Examples of functionality that shall not be added to ``scippneutron`` are handling of facility-specific file types or data layouts, or instrument-specific correction algorithms.
+
```
Comment (OwenArnold) [inline docs/about/contributing.rst:25]:
```suggestion
`ess <https://github.com/scipp/ess>`_ is an example codebase providing facility-specific algorithms
```

File: docs/about/release-notes.rst [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+Release Notes
+=============
+
+First release (planned)
+-----------------------
+
+Features
+~~~~~~~~
+
+* Functionality from ``scipp.neutron`` is not available in this package.
```
Comment (OwenArnold) [inline docs/about/release-notes.rst:12]: Expand on this maybe?

File: docs/conf.py [hunk 5]
```diff
@@ -14,29 +20,188 @@
...
+# You can specify multiple suffix as a list of string:
+#
+# source_suffix = ['.rst', '.md']
+source_suffix = '.rst'
+
+# The master toctree document.
+master_doc = 'index'
+
+# General information about the project.
+project = u'scipp'
```
Comment (OwenArnold) [inline docs/conf.py:73]: scippneutron

File: docs/index.rst [hunk 6]
```diff
@@ -1,20 +1,59 @@
...
 
-Toolkit for neutron scattering data reduction powered by scipp. Provides technique specific tools based on physics of neutron scattering.
+Toolkit for neutron scattering data reduction powered by `scipp <https://scipp.github.io/>`_.
+Provides "unit conversions" and technique specific tools based on physics of (time-of-flight) neutron scattering.
+
+Where can I get help?
+---------------------
+
+For questions not answered in the documentation
+`this page <https://github.com/scipp/scippneutron/issues?utf8=%E2%9C%93&q=label%3Aquestion>`_
```
Comment (OwenArnold) [inline docs/index.rst:11]: See above, usage of issues

File: docs/user-guide/groupby.ipynb [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+    "import scipp as sc"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "events = sc.neutron.load(filename='PG3_4844_event.nxs', load_pulse_times=False)"
```
Comment (OwenArnold) [inline docs/user-guide/groupby.ipynb:38]: These docs relate to "old" scipp + neutron package.

**Review by OwenArnold — CHANGES_REQUESTED** [review]

File: docs/about/contributing.rst (hunk 2 above)
Comment (SimonHeybrock) [reply docs/about/contributing.rst:11]: Yes, I think we should in the long run. If we truly want scipp to become generic and used by outside projects we should not have neutron/ess noise in there.

File: docs/about/release-notes.rst (hunk 4 above)
Comment (SimonHeybrock) [reply docs/about/release-notes.rst:12]:
```suggestion
* Functionality from ``scipp.neutron`` is now available in this package.
  This includes in particular the instrument view and "unit conversions" for time-of-flight neutron sources.
```

File: docs/about/contributing.rst (hunk 2 above)
Comment (OwenArnold) [reply docs/about/contributing.rst:11]: I agree about not having the noise. As a team we manage single set of  resources across multiple repositories, so we'll have to think about how we manage the backlogs and priorities on each of these repositories properly. I don't have a strong opinions about opening up issues on `scippneutron` for now.

File: docs/about/release-notes.rst [hunk 8]
```diff
@@ -9,7 +9,8 @@ First release (planned)
 Features
 ~~~~~~~~
 
-* Functionality from ``scipp.neutron`` is not available in this package.
+* Functionality from ``scipp.neutron`` is now available in this package.
```
Comment (OwenArnold) [inline docs/about/release-notes.rst:12]: 👍

**Comment by SimonHeybrock** [comment]
@OwenArnold I think I agressed your comments, but will be happy to address more in the inevitable follow-ups. Merging so we can see if this published correctly.

**Pull request merged**